include/sddf/i2c/queue.h

```diff
@@ -41,17 +41,23 @@ enum i2c_token {
     I2C_TOKEN_END = 0x0,
     /* START: Begin a transfer. Causes master device to capture bus. */
     I2C_TOKEN_START = 0x1,
-    /* ADDRESS WRITE: Used to wake up the target device on the bus.
+    /* ADDRESS WRITE: Address target and unset READ bit.
      * The byte immediately following this token is an integer (N) length of the succeeding
-       write. Max 256 (1**8). The next N bytes are the payload. */
+       write. Max 256 (1<<8). The next N bytes are the payload. */
     I2C_TOKEN_ADDR_WRITE = 0x2,
-    /* ADDRESS READ: Same as ADDRW but sets up DATA tokens as reads.
-       FIXME: N bytes must be padded after read to suit current transport layer. This should
-              be replaced, ideally by using a separate shared buffer for return data*/
+    /* ADDRESS READ: Address target and set READ bit. This is the final READ in a chain.
+     * The byte immediately following this token is an integer (N) length of the desired read.
+     * Max size 256.
+     */
     I2C_TOKEN_ADDR_READ = 0x3,
+    /* CONTINUING READ: same as ADDRESS READ, but doesn't end the read operation. Allows
+     * chaining of multiple reads for arbitrarily long read operations. A multi-read chain
+     * should consist of several READCs terminated by a final READ.
```

Comment:
> * should consist of several READCs terminated by a final READ.

This isn't checked anywhere?

Reply:
As discussed in person, we currently don't check any other possibly invalid conditions either because the I2C spec doesn't say this is invalid. If a client makes a bad request (i.e. that the target device can't understand), they will simply get no data back because the bus will time out quickly. It's also possible that the ability to represent the "wrong" condition here could be needed as a quirk on some devices.

Reply:
Still confused how this doesn't lead to a client just being able to monopolise the device but better to discuss in person.

Reply:
There's no way we can ever stop the client monopolising the device with the current design, irrespective of this change. I2C transactions have to happen all in one piece, so the only way we can implement something to the effect of fairness would be to implement a maximum transaction length. This change actually will make relevant interactions faster, not slower.

Reply:
Okay sure. The queue capacity should act as an implicit maximum transaction length.